### PR TITLE
Sort dependencies

### PR DESCRIPTION
**Pull request by basil**
### Problem

The dependencies in our `pom.xml` files are unsorted, making it difficult when editing the file to decide where to place new entries and also making edits prone to merge conflicts.

### Solution

Sort the dependencies by scope, then by group ID and artifact ID.

### Notes for reviewers

This PR looks scary, but it is trivial to review. I just edited

```xml
diff --git a/pom.xml b/pom.xml
index 88a20198a2..543310d4e7 100644
--- a/pom.xml
+++ b/pom.xml
@@ -588,6 +588,8 @@ THE SOFTWARE.
                 <lineSeparator>\n</lineSeparator>
                 <expandEmptyElements>false</expandEmptyElements>
                 <spaceBeforeCloseEmptyElement>true</spaceBeforeCloseEmptyElement>
+                <sortDependencies>scope,groupId,artifactId</sortDependencies>
+                <sortDependencyExclusions>groupId,artifactId</sortDependencyExclusions>
               </sortPom>
             </pom>
           </configuration>
```

and then ran `mvn spotless:apply`. This is the result. You only need to review the above change to `pom.xml`.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by alecharp — APPROVED** [review]

File: war/pom.xml [hunk 1]
```diff
@@ -64,16 +64,24 @@ THE SOFTWARE.
...
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>${project.groupId}</groupId>
+      <artifactId>remoting</artifactId>
+      <!-- specified in the parent -->
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.main</groupId>
       <artifactId>jenkins-core</artifactId>
```
Comment (daniel-beck) [inline war/pom.xml:78]: Should probably be between CLI and remoting but because `groupId` isn't the variable it's not?

File: test/pom.xml [hunk 2]
```diff
@@ -105,32 +98,33 @@ THE SOFTWARE.
...
+      <type>executable-war</type>
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>org.jenkins-ci.plugins</groupId>
-      <artifactId>cloudbees-folder</artifactId>
-      <version>6.708.ve61636eb_65a_5</version>
+      <groupId>io.jenkins.plugins</groupId>
+      <artifactId>javax-mail-api</artifactId>
+      <version>1.6.2-5</version>
```
Comment (daniel-beck) [inline test/pom.xml:110]: Would make more sense to group by plugin/non-plugins? Can this be done with this tool somehow?

File: bom/pom.xml [hunk 3]
```diff
@@ -358,88 +346,91 @@ THE SOFTWARE.
...
-      <!-- Modules -->
       <dependency>
-        <groupId>org.jenkins-ci.modules</groupId>
-        <artifactId>instance-identity</artifactId>
-        <version>2.2</version>
+        <groupId>org.slf4j</groupId>
+        <artifactId>log4j-over-slf4j</artifactId>
+        <version>${slf4jVersion}</version>
       </dependency>
+      <!-- SLF4J used in maven-plugin and core -->
```
Comment (daniel-beck) [inline bom/pom.xml:392]: This is out of place now (used to be above the block of all slf4j dependencies.

File: bom/pom.xml [hunk 4]
```diff
@@ -54,292 +61,273 @@ THE SOFTWARE.
...
       </dependency>
       <dependency>
-        <groupId>commons-httpclient</groupId>
-        <artifactId>commons-httpclient</artifactId>
-        <version>3.1-jenkins-3</version>
+        <groupId>com.sun.xml.txw2</groupId>
+        <artifactId>txw2</artifactId>
+        <version>20110809</version>
       </dependency>
+      <!--XStream-->
```
Comment (daniel-beck) [inline bom/pom.xml:101]: Used to introduce a block of two dependencies (xstream + kxml), now does not and makes no sense.

File: bom/pom.xml [hunk 5]
```diff
@@ -54,292 +61,273 @@ THE SOFTWARE.
...
       </dependency>
       <dependency>
-        <groupId>org.ow2.asm</groupId>
-        <artifactId>asm-analysis</artifactId>
-        <version>${asm.version}</version>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-compress</artifactId>
+        <version>1.21</version>
       </dependency>
+      <!--Groovy-->
```
Comment (daniel-beck) [inline bom/pom.xml:198]: Used two introduce a block of two dependencies (groovy and related jansi), now does not.

**Review by daniel-beck — DISMISSED** [review]
Some of the comments introducing a block of multiple dependencies are now out of place and just more confusing.

File: war/pom.xml (hunk 1 above)
Comment (basil) [reply war/pom.xml:78]: Correct

File: test/pom.xml (hunk 2 above)
Comment (basil) [reply test/pom.xml:110]: > Would make more sense to group by plugin/non-plugins?

Would it?

> Can this be done with this tool somehow?

No

File: bom/pom.xml (hunk 3 above)
Comment (basil) [reply bom/pom.xml:392]: Core doesn't import SLF4J, so the "and core" part of the comment was out of date before my PR and remains out of date after my PR.

> This is out of place now

No. `maven-plugin` only imports `org.slf4j.Logger` and `org.slf4j.LoggerFactory`, which are both defined in `slf4j-api-1.7.32.jar`. The comment is above the `slf4j-api` dependency, so it is not out of place.

File: bom/pom.xml (hunk 4 above)
Comment (basil) [reply bom/pom.xml:101]: Deleted

File: bom/pom.xml (hunk 5 above)
Comment (basil) [reply bom/pom.xml:198]: Deleted

**Comment by basil** [comment]
> Some of the comments introducing a block of multiple dependencies are now out of place and just more confusing.

I have deleted two comments that introduced a block of multiple dependencies and are now obsolete.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**